WordPress/src/main/java/org/wordpress/android/ui/qrcodeauth/QRCodeAuthUiState.kt

```diff
@@ -0,0 +1,204 @@
+        open val alpha: Float = BASE_ALPHA
+        open val primaryAction: Action? = null
+        open val secondaryAction: Action? = null
+        open val browser: String? = null
+        open val location: String? = null
+
+        data class Validated(
+            override val browser: String? = null,
+            override val location: String? = null,
+            override val primaryAction: ValidatedPrimaryAction,
+            override val secondaryAction: ValidatedSecondaryAction
+        ) : Content() {
+            override val type = VALIDATED
+            override val title: UiString = if (browser == null) {
+                UiStringResWithParams(R.string.qrcode_auth_flow_validated_title, listOf(UiStringText(location ?: " ")))
+            } else {
+                UiStringResWithParams(
+                        R.string.qrcode_auth_flow_validated_title,
+                        listOf(UiStringText(browser), UiStringText(location ?: " "))
+                )
+            }
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_validated_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_qrcode_auth_validated_152dp
+        }
+
+        data class Authenticating(
+            override val browser: String? = null,
+            override val location: String? = null,
+            override val primaryAction: AuthenticatingPrimaryAction,
+            override val secondaryAction: AuthenticatingSecondaryAction
+        ) : Content() {
+            override val type = AUTHENTICATING
+            override val title: UiString = if (browser == null) {
+                UiStringResWithParams(R.string.qrcode_auth_flow_validated_title, listOf(UiStringText(location ?: " ")))
+            } else {
+                UiStringResWithParams(
+                        R.string.qrcode_auth_flow_validated_title,
+                        listOf(UiStringText(browser), UiStringText(location ?: " "))
+                )
+            }
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_validated_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_qrcode_auth_validated_152dp
+            override val alpha: Float = BLURRED_ALPHA
+            override val isProgressShowing: Boolean = true
+        }
+
+        data class Done(
+            override val primaryAction: DonePrimaryAction,
+            override val secondaryAction: DoneSecondaryAction
+        ) : Content() {
+            override val type = DONE
+            override val title: UiString = UiStringRes(R.string.qrcode_auth_flow_done_title)
+            override val subtitle: UiString = UiStringRes(R.string.qrcode_auth_flow_done_subtitle)
+            @DrawableRes override val image = R.drawable.img_illustration_qrcode_auth_login_success_218dp
+        }
+    }
+
+    sealed class Action {
+        open val label: UiString? = null
+        open val isEnabled: Boolean = true
```

Comment:
question ❓: I couldn't understand what the purpose of this `isEnabled` and `isVisible` field is for?. Could you clarify?. I couldn't associate these fields with the `Action` class. Perhaps these are for a button? In that case, do you think we can have a better name than `Action`?.

Reply:
Thanks @AjeshRPai . Yes, this is a button. The button can be shown on the view, but disabled when the progress view is shown on top of it. Would love to hear your thoughts re: different name ... `ActionButton` ??

Reply:
I went with ActionButton. :)